crates/rattler_conda_types/src/minimal_prefix_record.rs

```diff
@@ -0,0 +1,298 @@
+//! Minimal prefix record reading for fast environment change detection.
+//!
+//! This module provides functionality to read only the minimal metadata needed
+//! from conda-meta JSON files to determine if packages have changed, avoiding
+//! the expensive parsing of file lists and other large data structures.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use crate::{NoArchType, PackageName, PackageRecord, PrefixRecord, Version, VersionWithSource};
+use hex;
+use rattler_digest::{Md5Hash, Sha256Hash};
+
+/// A minimal version of `PrefixRecord` that only contains fields needed for transaction computation.
+/// This is much faster to parse than the full `PrefixRecord`.
+#[derive(Debug, Clone)]
+pub struct MinimalPrefixRecord {
```

Comment:
Looking at `Transaction::from_current_and_desired` you need waaaaay less fields! 

I think you only need:

- `name`, `version`, `build` but these can be derived from the name of the file. Because its `<name>-<version>-<build>.json`. 

```rust
// Filename is in the form of: <name>-<version>-<build>
let (build_string, version, name) = filename_without_ext.rsplitn(3, '-').next_tuple()?;
```

- `python_site_packages_path` only if the `name` is `"python"`.
- `sha256` 
- `md5` but only if there is no `sha256`
- `size` but only if there is no `sha256` and no `md5`.

Reply:
I think this warrants a specialized path for this particular case.